client/src/components/Visualization/charts/Geochart.js

```diff
@@ -4,45 +4,74 @@
  * Date: 2020/07/13
  */
 
-import { Chart } from 'react-google-charts';
-import React from 'react';
-import { ads } from '../../../firebase/FirestoreDocumentReader';
-import firebase from '../../../firebase/firebase';
-import { states } from './StateDataParser';
+import { Chart } from "react-google-charts";
+import React, { useState, useEffect } from 'react';
+import { STATES } from '../../../constants/geochart_constants';
+import { database } from '../../../firebase/firebase';
 
 const Geochart = () => {
-  /*
-   * Currently, getData assigns a meaningless random number to each state.
-   * In the future, once the states dictionary contains ad information, getData will
-   *    retrieve information such as minimum ad impressions and maximum ad spend.
-   */
-
-  function getData() {
-    const data = [['State', 'Random Number']];
-    for (const state in states) {
-      data.push([state, Math.floor(Math.random() * Math.floor(1000))]);
+
+  const [adTotal, setAdTotal] = useState(0);
+  const WIDTH = "700";
+  const HEIGHT = "400";
+
+  useEffect(() => {
+    async function fetchStateTotals() {
+
+      let data = [["State", "Total Ad Spend (USD)", {type: 'string', role: 'tooltip'}]];
+
+      for (let state of STATES) {
+
+        // Calculate state spend totals.
+        let stateTotal = 0;
+        let stateSpendData = await database.collection('dev_states')
+                                           .doc(state.toLowerCase())
+                                           .collection('stateData')
+                                           .doc('spendData')
+                                           .get();
```

Comment:
How big is this query? Reason I'm asking is bc it seems like you only want one field, in which case, why not just have a separate document for that field so you don't have to wait for `n` things to load?

Reply:
That was a good suggestion, thank you. I've moved totalStateSpend up so that it's now a field of the top level state document & refactored the query accordingly. The change shows up in the "Files Changed" but not in this comment, so I'm resolving to show that it's been addressed.